### ISSUE TITLE
Widget Fill Screen option reverting back to being Disabled every morning..

### ISSUE THREAD

**Issue by Andorski** [8bitPit/Niagara-Issues#2153]
### Describe the bug

Every morning the custom widget that I previously set to Fill Screen gets reverted back to not filling the screen.  To set it back I have to select Disable in the Fill Screen option first (even though it's already been disabled) before re-enabling it. After re-enabling the option the widget is properly filling the screen again - until the next morning.

I'm using a custom KWGT widget. Have the Pro version of Niagara. On a Pixel 7 Pro. Already tried to Restart Launcher in the Advanced Options menu.

### Steps to reproduce

_Steps 1-3 are only if the Fill Screen option isn't already enabled.  Otherwise skip to step 4._

1. Hold Press on the custom widget to open options menu.
2. Select Fill Screen.
3. Select Enable.
4. Wait until next morning when the custom widget stops filling the screen.
5. Hold Press on the custom widget to open options menu.
6. Select Fill Screen.
7. Select Disable.
8. Hold Press on the custom widget to open options menu.
9. Select Fill Screen.
10. Select Enable.
11. Repeat steps 4-10.

### Does the issue also occur with other 3rd-party launchers?

Not applicable

### Device name

Google Pixel 7 Pro

### Android version

Android 13

**Comment by Amorphous404** [comment]
There were a few more reports on Telegram, I'll attach them here:


Posted by TJ
November 20 2022

![IMG_20221216_151908_623.jpg](https://user-images.githubusercontent.com/96917972/208071492-8cb12ad9-ed6f-473e-8edd-412dbf8ac7b1.jpg)

For some reason after a while the Widget resizes itself a few times per day (Fill screen is on). But I have to resize it.

Other similar widgets scale randomly too.

OnePlus Nord
OxygenOS
Android 12
App: Today Weather
Niagara v1.7.10

The behaviour started after 1.7.10, haven't been earlier.

**Comment by Amorphous404** [comment]
Posted By Brain Evers
November 10 2022

Niagara v1.8.1
App: Weawow Weather Widget

![IMG_20221216_153244_229.jpg](https://user-images.githubusercontent.com/96917972/208074111-ed17dfc5-e291-4a4e-b250-b912547720f7.jpg)
Has anyone had the fill screen option for widgets stop working? Randomly I'll notice.my widget is back to normal size instead of filling the screen. I have to select my widget, disable fill screen, then enable fill screen again for it to go back.


![IMG_20221216_152551_908.jpg](https://user-images.githubusercontent.com/96917972/208073003-0aa57fab-18d5-4a38-986c-52e039475df5.jpg)


Me:
Thanks for reporting it 👍, can you also try to use a different widget for a while and check if the same issue persists with other widgets?

Brain:
The same thing happened using a different weather widget. Not sure what triggered it. It was fine when I went to bed. I woke up this morning and the widget was back to the unfilled screen size.

Me:
Btw forgot to ask you one thing, does restarting the launcher restore back the size?

Brain:
Again in the morning the widget had gone back to the unfilled size. **I force stopped Niagara and relaunched it and the widget was back to the filled size. So restarting the launcher fixes it.**

**Comment by Amorphous404** [comment]
Posted by Nonchalant FTP
April 10 2022

![IMG_20221216_154129_896.jpg](https://user-images.githubusercontent.com/96917972/208075647-1ca710e2-37cc-4d51-8329-38180ef986ec.jpg)
![IMG_20221216_154133_007.jpg](https://user-images.githubusercontent.com/96917972/208075682-1bcae21c-3214-4c3b-8d67-70921b155543.jpg)

Have y'all experienced the issue where "fill screen" doesn't stick?

Sometimes kwgt reverts back to unfilled screen

**Comment by Amorphous404** [comment]
Hey @Andorski does restarting the Launcher helps? (Niagara settings>Advanced>Restart Launcher)

**Comment by Andorski** [comment]
> Hey @Andorski does restarting the Launcher help? (Niagara settings>Advanced>Restart Launcher)

Restarted the launcher last morning, which initially did set the widget back to full screen.  But again this morning I found that the widget reverted back to not filling the screen.

**Comment by tjsuominen** [comment]
Yeah, this still persist every morning when I wake up... Restarting Niagara / phone does not help.

**Comment by Andorski** [comment]
> Yeah, this still persist every morning when I wake up... Restarting Niagara / phone does not help.

What phone and what version of Android are you using?

**Comment by tjsuominen** [comment]
OP Nord with Android 12 (OOS 12.1).
